File: src/client/java/com/rafalohaki/event/Event.java
package com.rafalohaki.event;

/**
 * Base class for all events.
 */
public abstract class Event {
}

/**
 * Cancellable event base class.
 */
public abstract class CancellableEvent extends Event {
    private boolean cancelled = false;
    
    public boolean isCancelled() {
        return cancelled;
    }
    
    public void cancel() {
        cancelled = true;
    }
}

File: src/client/kotlin/com/rafalohaki/event/Event.kt
package com.rafalohaki.event

/**
 * Base class for all events.
 */
abstract class Event

/**
 * Cancellable event base class.
 */
abstract class CancellableEvent : Event() {
    var cancelled = false
        private set
    
    fun cancel() {
        cancelled = true
    }
}